app/src/main/java/io/rafaelribeiro/spendless/navigation/RootAppNavigation.kt

```diff
@@ -291,9 +295,6 @@ fun RootAppNavigation(
                 onEvent = viewModel::onEvent,
                 launchedFromWidget = launchedFromWidget
             )
-            LaunchedEffect(key1 = Unit) {
-                mainViewModel.startSession()
-            }
```

Comment:
I removed this launch effect and called `startSession()` explicitly from different places as stated in requirements (after login, prompt screen & registration flow)